types/node/stream.d.ts

```diff
@@ -307,7 +307,12 @@ declare module "stream" {
             ): Promise<void>;
         }
 
-        interface Pipe { }
+        interface Pipe {
+            close(): void;
+            hasRef(): boolean;
+            ref(): void;
+            unref(): void;
```

Comment:
I think `ref()` and `unref()` usually return `this`.
Not sure if all variants of Pipes/Handles used in child process use case actually have a `close()` API and I'm even not sure if it is a good idea to add them here as NodeJs doc tells `Accessing the IPC channel fd in any way other than process.send() or using the IPC channel with a child process that is not a Node.js instance is not supported.` (see [doc](https://nodejs.org/dist/latest-v13.x/docs/api/child_process.html#child_process_options_stdio) point 2).
In spawned process there is `process.disconnect()` to close `process.channel` (see [docs](https://nodejs.org/dist/latest-v13.x/docs/api/process.html#process_process_disconnect)) but it seems there is no `disconnect` on parent side.

Reply:
> I think `ref()` and `unref()` usually return `this`.

In the case of this `Pipe` type they don't seem to.

> Accessing the IPC channel fd in any way other than ...

I don't think this applies because none of these method return the `fd`.

Reply:
well, they don't return the `fd` but I'm quite sure `close()` interacts with it.

Anyhow, I assume you use this in your app and it works there so it should be fine.